### PR TITLE
enhance: add addition index params for raft index

### PR DESCRIPTION
**Pull request by cqy123456**
issue: https://github.com/milvus-io/milvus/issues/29230

### CODE REVIEW

**Comment by cqy123456** [comment]
rerun ut

**Review by yah01 — APPROVED** [review]
/lgtm

**Comment by cqy123456** [comment]
rerun ut

File: pkg/util/indexparamcheck/constraints.go [hunk 1]
```diff
@@ -51,7 +57,7 @@ var (
 	BinIvfMetrics             = []string{metric.HAMMING, metric.JACCARD}                                             // const
 	HnswMetrics               = []string{metric.L2, metric.IP, metric.COSINE, metric.HAMMING, metric.JACCARD}        // const
 	RaftMetrics               = []string{metric.L2, metric.IP}
-	CagraMetrics              = []string{metric.L2}                                              // const
+	CagraBuildAlgoType        = []string{CargaBuildAlgoIVFPQ, CargaBuildAlgoNNDESCENT}
```
Comment (longjiquan) [inline pkg/util/indexparamcheck/constraints.go:60]: Maybe `CagraBuildAlgoTypes` is better.

File: pkg/util/indexparamcheck/raft_ivf_pq_checker.go [hunk 2]
```diff
@@ -57,6 +57,13 @@ func (c *raftIVFPQChecker) checkPQParams(params map[string]string) error {
 	if dimension%m != 0 {
 		return fmt.Errorf("dimension must be able to be divided by `m`, dimension: %d, m: %d", dimension, m)
 	}
+
+	setDefaultIfNotExist(params, RaftCacheDatasetOnDevice, "false")
+
+	if !CheckStrByValues(params, RaftCacheDatasetOnDevice, []string{"true", "false"}) {
+		return fmt.Errorf("raft index cache_dataset_on_device param only support true false")
```
Comment (longjiquan) [inline pkg/util/indexparamcheck/raft_ivf_pq_checker.go:64]: Add ut for this case.

**Review by longjiquan — CHANGES_REQUESTED** [review]

**Review by longjiquan — COMMENTED** [review]
/lgtm

**Pull request merged**